### DISCUSSION TITLE
How generating a key via hashFiles is supposed to work if the cache is not yet restored?

### DISCUSSION THREAD

**Discussion by leshikus** [actions/cache#791] [Q&A] [answered]
I wonder how the standard usecase with key =  ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }} works. The cache is not restored yet, how hashFiles finds the files to hash?

**Comment by dhadka** [comment, marked as answer]
The idea is that you're computing the hash of files already checked into the repo.  For the case of `package-lock.json`, NPM recommends checking in this file as it contains the exact dependencies and versions used by your program (more details at https://docs.npmjs.com/cli/v8/configuring-npm/package-lock-json).

**Reply by leshikus** [reply]
Thanks!
